fix: updated existing enzyme tests to use react testing library

jciasenza opened the pull request:
### Description
This PR replaces Enzyme with React Testing Library (RTL) to improve test reliability and ensure compatibility with React 18. Enzyme is deprecated and no longer recommended for testing React applications.

**Changes Made**
1 - **Replaced Enzyme with RTL** – Aligns with modern testing practices.
2 - **Mocked** `BlockBrowserContainer` – Improves test isolation by removing external dependencies.
3 - **Rendered Main inside a Redux provider** – Simulates the application state with mock props.
4 - **Added** `reportDownloadEndpoint` prop – Ensures the component receives all necessary props.
5 - **Updated dropdown interaction tests:**
 - Loading course blocks: Verified `fetchCourseBlocksMock` is called with courseId and `excludedBlockTypes`.
 - Dropdown menu visibility: Ensured it appears when clicked and disappears when clicked again.

### Other information
**Dependencies**
- Added (RTL)
@testing-library/react@12.1.5
@testing-library/jest-dom@6.6.3
@testing-library/user-event@12.8.3

- Removed (Enzyme)
enzyme@3.11.0
react-16-enzyme-adapter@1.15.8
jest-enzyme@7.1.2

openedx-webhooks commented:
Thanks for the pull request, @jciasenza!


This repository is currently maintained by `@openedx/wg-maintenance-edx-platform`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary>:radio_button: Get product approval</summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details>

<details><summary>:radio_button: Provide context</summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details>





<details><summary>:radio_button: Get a green build</summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>


---

<details><summary>Where can I find more information?</summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details>

<details><summary>When can I expect my changes to be merged?</summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>
<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

jciasenza commented:
@brian-smith-tcril  check this PR, thacks

brian-smith-tcril commented on lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.test.jsx:
```diff
@@ -1,96 +1,99 @@
+            expect(screen.queryByTestId('mocked-block-browser-container')).toBeNull();
+            await act(async () => {
+                await userEvent.click(screen.getByText('Select a section or problem'));
+            });
```
Does this need to be wrapped in `await act`?

What happens when this is just
```suggestion
            await userEvent.click(screen.getByText('Select a section or problem'));
```

jciasenza replied:
Yes it could be, I have tried it and it does not give errors. Also remove act on import

brian-smith-tcril commented on lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.test.jsx:
```diff
@@ -1,96 +1,99 @@
+            ).toHaveClass('block-browser'));
+            await act(async () => {
+                await userEvent.click(screen.getByText('Select a section or problem'));
+            });
```
Does this need to be wrapped in `await act`?

What happens when this is just
```suggestion
            await userEvent.click(screen.getByText('Select a section or problem'));
```

jciasenza replied:
Yes it could be, I have tried it and it does not give errors. Also remove act on import

brian-smith-tcril reviewed (approved):
This LGTM! Thanks for working through this and making sure all the tests are properly reimplemented using RTL!

I left a couple comments with some small questions in there, but I consider those non-blocking.

jciasenza commented:
Without conflicts, this is the cleaning of enzyme dependencies.

brian-smith-tcril commented:
@jciasenza I'm not sure what you mean by
> Without conflicts

As of writing I see
![image](https://github.com/user-attachments/assets/46cfa9c2-c00b-4d42-8dc5-bdef5a6f63a8)

jciasenza commented:
> @jciasenza I'm not sure what you mean by
> 
> > Without conflicts
> 
> As of writing I see ![image](https://private-user-images.githubusercontent.com/112954497/427584754-46cfa9c2-c00b-4d42-8dc5-bdef5a6f63a8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMwODQwMDEsIm5iZiI6MTc0MzA4MzcwMSwicGF0aCI6Ii8xMTI5NTQ0OTcvNDI3NTg0NzU0LTQ2Y2ZhOWMyLWMwMGItNGQ0Mi04ZGM1LWJkZWY1YTZmNjNhOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMyN1QxMzU1MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NWZjYTQ0YjE3YjY5MTQ5MGMyMzFmNzczYTM3NWQ0ZDNjMTc3ZTBlNjlhZmQ0N2ZlMTk3ZWNhMmQ5OGViMzAxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.T6z4Xcw0yR8yh0Qv9q0KkQV2yymxYLeKM3uFEcEWPjs)

What I mean is that the conflicts that appear are the dependencies that I added from RTL and those that I removed from Enzyme, because everything else is ok

brian-smith-tcril commented:
@jciasenza this branch needs to be rebased on latest `master`. It cannot be merged until there are no merge conflicts.

jciasenza commented:
> @jciasenza this branch needs to be rebased on latest `master`. It cannot be merged until there are no merge conflicts.

Ok @brian-smith-tcril, now I'll update the branch and let you know.

Commit:
fix: updated existing enzyme tests to use react testing library

jciasenza commented:
@brian-smith-tcril, already updated the branch I think it's ok now, for merged. Thanks !

feanil committed:
Merge branch 'master' into jciasenza
